src/riscv/lib/src/state_backend/region.rs

```diff
@@ -369,14 +369,16 @@ impl<E: Decode<()>, const LEN: usize, M: ManagerDeserialise> Decode<()> for Cell
     }
 }
 
-impl<A: PartialEq<B> + Copy, B: Copy, const LEN: usize, M: ManagerRead, N: ManagerRead>
+impl<A: PartialEq<B>, B, const LEN: usize, M: ManagerRead, N: ManagerRead>
     PartialEq<Cells<B, LEN, N>> for Cells<A, LEN, M>
 {
     fn eq(&self, other: &Cells<B, LEN, N>) -> bool {
-        (0..LEN).all(|i| self.read(i) == other.read(i))
+        (0..LEN).all(|i| self.read_ref(i).eq(other.read_ref(i)))
     }
 }
 
+impl<T: Eq + Copy, const LEN: usize, M: ManagerRead> Eq for Cells<T, LEN, M> {}
```

Comment:
is this copy bound necessary?
```suggestion
impl<T: Eq, const LEN: usize, M: ManagerRead> Eq for Cells<T, LEN, M> {}
```

Reply:
It's not! Good catch. Removed the redundant trait bound in [this diff](https://github.com/tezos/riscv-pvm/compare/ed8acdf9674ba25d687ad3120b7174375f75b866..5316e45f6daca221c33ce2d7d08cfdfd044c4662).